### PR TITLE
fix(col): fix strange style of layout

### LINKED ISSUE #872: Discuss about layout style
Comparing several frameworks, element layout has a strange style. Does this need improvement?

image

```html
<h3>Element-plus</h3>
<el-row :gutter="20">
    <el-col :span="12">
        <div class="item" style="height: 50px">1</div>
    </el-col>
    <el-col :span="12">
        <div class="item" style="height: 30px">2</div>
    </el-col>
    <el-col :span="6">
        <div class="item" style="height: 40px">3</div>
    </el-col>
    <el-col :span="6">
        <div class="item" style="height: 60px">4</div>
    </el-col>
    <el-col :span="12">
        <div class="item" style="height: 30px">5</div>
    </el-col>
</el-row>

<h3>Vant</h3>
<van-row :gutter="20">
    <van-col :span="12">
        <div class="item" style="height: 50px">1</div>
    </van-col>
    <van-col :span="12">
        <div class="item" style="height: 30px">2</div>
    </van-col>
    <van-col :span="6">
        <div class="item" style="height: 40px">3</div>
    </van-col>
    <van-col :span="6">
        <div class="item" style="height: 60px">4</div>
    </van-col>
    <van-col :span="12">
        <div class="item" style="height: 30px">5</div>
    </van-col>
</van-row>

<h3>Antd Vue</h3>
<a-row :gutter="20">
    <a-col :span="12">
        <div class="item" style="height: 50px">1</div>
    </a-col>
    <a-col :span="12">
        <div class="item" style="height: 30px">2</div>
    </a-col>
    <a-col :span="6">
        <div class="item" style="height: 40px">3</div>
    </a-col>
    <a-col :span="6">
        <div class="item" style="height: 60px">4</div>
    </a-col>
    <a-col :span="12">
        <div class="item" style="height: 30px">5</div>
    </a-col>
</a-row>
```

### PR DESCRIPTION
**Pull request by kooriookami**
see issue

BREAKING CHANGE: Each single col item will fill the row

fix #872

Please make sure these boxes are checked before submitting your PR, thank you!

* [ ] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/theme-chalk/src/col.scss [hunk 1]
```diff
@@ -12,7 +12,8 @@
 
 @for $i from 0 through 24 {
   .el-col-#{$i} {
-    width: (1 / 24 * $i * 100) * 1%;
+    max-width: (1 / 24 * $i * 100) * 1%;
+    flex: 0 0 (1 / 24 * $i * 100) * 1%;
```
Comment (JeremyWuuuuu) [inline packages/theme-chalk/src/col.scss:16]: Adding flex seems weird since this is not a child of `flex` displayed HTML Element

Comment (kooriookami) [reply packages/theme-chalk/src/col.scss:16]: I refer to other frameworks, they also use `flex`.

Comment (JeremyWuuuuu) [reply packages/theme-chalk/src/col.scss:16]: So could you give me a screen shot on this?

Comment (JeremyWuuuuu) [reply packages/theme-chalk/src/col.scss:16]: We closed testing deploy for saving the build time.. So I cannot preview the changes. Thank you

Comment (kooriookami) [reply packages/theme-chalk/src/col.scss:16]: Yes, I also create a preview website, I will delete it after the pr is closed.
image

https://test.kooriookami.top/deploy-preview-875--element-plus/#/zh-CN/component/layout

Comment (JeremyWuuuuu) [reply packages/theme-chalk/src/col.scss:16]: LGTM

**Review by jw-foss — APPROVED** [review]

**Pull request merged**